iceoryx2-pal/posix/src/linux/errno.rs

```diff
@@ -151,6 +151,7 @@ impl Errno {
     }
 }
 
+#[cfg(not(all(test, loom, feature = "std")))]
```

Comment:
Curious, why is this removed for the loom build?

Reply:
This function contains static variable `IS_LOCKED`, which loom can't emulate. Since this is top level function and not used in the internal code path, I just exclude that function from loom instead of doing this:

```rust
pub unsafe fn strerror_r() {
  ...
  #[cfg(all(test, loom, feature = "std"))]
  static IS_LOCKED: IoxAtomicBool = IoxAtomicBool::new(false);
  #[cfg(not(all(test, loom, feature = "std")))]
  static IS_LOCKED: std::sync::LazyLock<IoxAtomicU32> = std::sync::LazyLock::new(|| { unimplemented!("loom does not provide const-initialization for atomic variables.") });
  ...
}
```

... but this might be confusing, as all other places do this way. I may update this to unify the style.